packages/top-app-bar/index.js

```diff
@@ -67,6 +77,17 @@ export default class TopAppBar extends React.Component {
     return React.cloneElement(element, {hasRipple: true});
   }
 
+  setStyle = (varName, value) => {
+    const updatedStyle = Object.assign({}, this.state.style);
+    updatedStyle[varName] = value;
+    this.setState({style: updatedStyle});
+  }
+
+  getMergedStyles = () => {
+    const {style} = this.state;
+    return Object.assign({}, style, this.props.style);
```

Comment:
Is there a reason we don't want to add `props.style` to the `state`? What if we add `props.style` to the state in the constructor, and just return `state.style` here?

Reply:
I thought this was the simplest approach. We update this.state.style in the setStyle adapter method. So every time we call setStyle, we would need to merge this.props.style. Also if this.props.style every updates, we would need to merge it again with this.state.style.

I think doing it in this method is the most straightforward.

Reply:
Oh I see, this makes sense if we don't want to have to keep track of whether or not props.style was merged into the state.